diff --git a/ocaml/examples/Makefile.am b/ocaml/examples/Makefile.am
index 14e534b3c..771f5740b 100644
--- a/ocaml/examples/Makefile.am
+++ b/ocaml/examples/Makefile.am
@@ -48,7 +48,7 @@ if HAVE_OCAML
 
 noinst_SCRIPTS = create_disk debug_logging inspect_vm
 
-OCAMLFINDFLAGS = -cclib -L$(top_builddir)/lib/.libs
+OCAMLFINDFLAGS = -cclib '-L$(top_builddir)/lib/.libs -L$(top_builddir)/gnulib/lib/.libs'
 
 if HAVE_OCAMLOPT
 create_disk: create_disk.ml
