diff --git a/ocaml/Makefile.am b/ocaml/Makefile.am
index b31e8fc65..b442bb31f 100644
--- a/ocaml/Makefile.am
+++ b/ocaml/Makefile.am
@@ -67,13 +67,15 @@ stamp-mlguestfs: libguestfsocaml.a $(guestfs_cmm)
 	  -ldopt '$(LDFLAGS)' \
 	  $(libguestfsocaml_a_OBJECTS) guestfs.cmo \
 	  $(LTLIBINTL) \
-	  -L../lib/.libs -lguestfs
+	  -L../lib/.libs -lguestfs \
+	  -L../gnulib/lib/.libs -lgnu
 if HAVE_OCAMLOPT
 	$(OCAMLMKLIB) -o mlguestfs \
 	  -ldopt '$(LDFLAGS)' \
 	  $(libguestfsocaml_a_OBJECTS) guestfs.cmx \
 	  $(LTLIBINTL) \
-	  -L../lib/.libs -lguestfs
+	  -L../lib/.libs -lguestfs \
+	  -L../gnulib/lib/.libs -lgnu
 endif
 	touch $@
 
