diff --git a/diskimage_builder/elements/debootstrap/root.d/08-debootstrap b/diskimage_builder/elements/debootstrap/root.d/08-debootstrap
index 87f90ecf..6650bd91 100755
--- a/diskimage_builder/elements/debootstrap/root.d/08-debootstrap
+++ b/diskimage_builder/elements/debootstrap/root.d/08-debootstrap
@@ -56,7 +56,7 @@ else
         DEBOOTSTRAP_QEMU="qemu-"
     fi
 
-    sudo sh -c "http_proxy=$http_proxy no_proxy=$no_proxy ${DEBOOTSTRAP_QEMU}debootstrap --verbose \
+    sudo env "DEBOOTSTRAP_DIR=$DEBOOTSTRAP_DIR" "PATH=$PATH" sh -c "http_proxy=$http_proxy no_proxy=$no_proxy ${DEBOOTSTRAP_QEMU}debootstrap --verbose \
         --variant=minbase \
         --components=${DIB_DEBIAN_COMPONENTS} \
         --arch=${ARCH} \
