diff --git a/m4/guestfs-qemu.m4 b/m4/guestfs-qemu.m4
index 62616ffff..c959d3650 100644
--- a/m4/guestfs-qemu.m4
+++ b/m4/guestfs-qemu.m4
@@ -41,7 +41,7 @@ AS_IF([test "x$with_qemu" = "xno"],[
         [$PATH$PATH_SEPARATOR/usr/sbin$PATH_SEPARATOR/sbin$PATH_SEPARATOR/usr/libexec])
     test "x$QEMU" = "xno" && AC_MSG_ERROR([qemu must be installed])
 
-    AC_DEFINE_UNQUOTED([QEMU],["$QEMU"],[Location of qemu binary.])
+    AC_DEFINE_UNQUOTED([QEMU],["/snap/octavia-diskimage-retrofit/current/usr/bin/qemu-system-$QEMU_CPU"],[Location of qemu binary.])
 
     dnl Check that the chosen qemu has virtio-serial support.
     dnl For historical reasons this can be disabled by setting
